packages/cli/src/cmds/validator/voluntaryExit.ts

```diff
@@ -115,39 +115,69 @@ ${validatorsToExit.map((v) => `${v.pubkey} ${v.index} ${v.status}`).join("\n")}`
       }
     }
 
-    for (const [i, {index, signer, pubkey}] of validatorsToExit.entries()) {
-      const slot = computeStartSlotAtEpoch(exitEpoch);
-      const domain = config.getDomainForVoluntaryExit(slot);
-      const voluntaryExit: phase0.VoluntaryExit = {epoch: exitEpoch, validatorIndex: index};
-      const signingRoot = computeSigningRoot(ssz.phase0.VoluntaryExit, voluntaryExit, domain);
-
-      let signature;
-      switch (signer.type) {
-        case SignerType.Local:
-          signature = signer.secretKey.sign(signingRoot);
-          break;
-        case SignerType.Remote: {
-          const signatureHex = await externalSignerPostSignature(config, signer.url, pubkey, signingRoot, slot, {
-            data: voluntaryExit,
-            type: SignableMessageType.VOLUNTARY_EXIT,
-          });
-          signature = bls.Signature.fromHex(signatureHex);
-          break;
+    const alreadySubmitted = [];
+    for (const [i, validatorToExit] of validatorsToExit.entries()) {
+      const {err} = await wrapError(processVoluntaryExit({config, client}, exitEpoch, validatorToExit));
```

Comment:
If we wanted to support better logging, esp in the case an exit is already processed by the chain, we could modify `isValidVoluntaryExit` to return an enum value

```ts
export enum VoluntaryExitValidity {
  valid = "valid",
  inactive = "inactive",
  alreadyExited = "already_exited",
  earlyEpoch = "early_epoch",
  shortTimeActive = "short_time_active"
  invalidSignature = "invalid_signature"
}

export function isValidVoluntaryExit(
  state: CachedBeaconStateAllForks,
  signedVoluntaryExit: phase0.SignedVoluntaryExit,
  verifySignature = true
): boolean {
  const {config, epochCtx} = state;
  const voluntaryExit = signedVoluntaryExit.message;
  const validator = state.validators.get(voluntaryExit.validatorIndex);
  const currentEpoch = epochCtx.epoch;

  // verify the validator is active
  if(isActiveValidator(validator, currentEpoch) return VoluntaryExitValidity.inactive;
  ...

  return VoluntaryExitValidity.valid;
}
```

That enum value can get passed up into an error message / code and checked against for logging.

Reply:
Ok, lets make a low priority issue for it based on @barnabasbusa 's comment of:

> Interesting to see that one of the validators (that have already exited) reports to be: 1/100: Internal Server Error: VOLUNTARY_EXIT_ERROR_INVALID

I read that "interesting to see" as "its weird/confusing/bad ux to see"

Reply:
It's not great UX to show internal server error imho, but I'm happy the bulk exists just work! Happy to get it merged as is.

Reply:
> It's not great UX to show internal server error imho

The beacon node returning more user friendly errors could solve this as well, [as per spec](https://ethereum.github.io/beacon-APIs/#/Beacon/submitPoolVoluntaryExit) we should return a 400 (and not 500) error if voluntary exit is invalid.